### ISSUE TITLE
[Bug]: textContent should be used as aria-label when no label attribute is provided in meter and progressbar components

### ISSUE THREAD

**Issue by dineshy87** [adobe/spectrum-web-components#3146]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

ProgressBar, Meter

### Expected behavior

To improve accessibility, the component should be updated to set the aria-label attribute based on the following conditions:

1. If only label is provided, use the label attribute value as both the aria-label and the visible label (For ProgressBar and Meter).
2. If only textContent is provided, use the textContent as both the aria-label and the visible label for progress bar and Meter. For ProgressCircle use it to update just aria-label since there is no visible label.
3. If both label and textContent are provided, use the label attribute value as the aria-label, and display textContent as the visible label (in the sp-field-label element).

### Actual behavior

Currently, the <sp-meter> component sets the aria-label attribute only if the label attribute is provided. When only textContent is provided, the component doesn't set the aria-label, leading to an accessibility issue flagged by Axe DevTools.

For example, try: 

`<sp-meter size="m" progress="71">Tasks Completed</sp-meter>`

The same issue can be observed in examples provided in docs for Meter component: https://opensource.adobe.com/spectrum-web-components/components/meter/

I also created similar combinations with ProgressBar and ProgressCircle here: The above issue can be observed at https://studio.webcomponents.dev/edit/H0daZwrqfYXamkVzquST/src/index.stories.js?p=stories 

Please run Axe DevTools, and observe the errors.

### How can we reproduce this issue?

3. Scroll to '....'
4. Check console
5. See error

**Comment by Rajdeepc** [comment]
@TarunAdobe Please add [this](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd) chrome extension to verify the above scenarios. FYI

**Comment by TarunAdobe** [comment]
Yus Yus I have done it already. Thnx

**Comment by TarunAdobe** [comment]
PR -> #3483
